lib/cucumber/events/bus.rb

```diff
@@ -8,79 +10,31 @@ module Events
     #
     # @private
     class Bus
-
-      def initialize(default_namespace)
-        @default_namespace = default_namespace.to_s
-        @handlers = {}
+      def initialize(resolver)
+        @resolver = resolver
+        @handlers = Hash.new { |h, k| h[k] = [] }
       end
 
       # Register for an event
       def register(event_id, handler_object = nil, &handler_proc)
         handler = handler_proc || handler_object
-        raise ArgumentError.new("Please pass either an object or a handler block") unless handler
-        event_class = parse_event_id(event_id)
-        handlers_for(event_class) << handler
-      end
 
-      # Broadcast an event
-      def notify(event)
-        handlers_for(event.class).each { |handler| handler.call(event) }
-      end
+        raise ArgumentError, "Please pass either an object#call or a handler block" if handler.nil? || !handler.respond_to?(:call)
 
-      private
+        event_class = @resolver.transform(event_id)
 
-      def handlers_for(event_class)
-        @handlers[event_class.to_s] ||= []
-      end
+        fail EventNameResolveError, %(Transforming "#{event_id}" into an event name failed for unknown reason.) if event_class.nil?
```

Comment:
No. Since we support user defined NameResolvers, this is a countermeasure against "NameResolvers".